src/main/kotlin/de/timo_reymann/ansible_vault_integration/intention/menu/VaultFileMenuAction.kt

```diff
@@ -5,57 +5,49 @@ import com.intellij.openapi.actionSystem.AnActionEvent
 import com.intellij.openapi.actionSystem.LangDataKeys
 import com.intellij.openapi.progress.ProgressManager
 import com.intellij.psi.PsiFile
+import com.intellij.psi.PsiManager
 import de.timo_reymann.ansible_vault_integration.config.AnsibleConfigurationService
 import de.timo_reymann.ansible_vault_integration.execution.AnsibleVaultTask
 import de.timo_reymann.ansible_vault_integration.intention.AnsibleVaultIdentityPopup
 import de.timo_reymann.ansible_vault_integration.runnable.AnsibleVaultRunnable
 import de.timo_reymann.ansible_vault_integration.runnable.DecryptFileAnsibleVaultRunnable
 import de.timo_reymann.ansible_vault_integration.runnable.EncryptFileAnsibleVaultRunnable
-import org.jetbrains.yaml.psi.YAMLFile
-import setVisible
 
 open class VaultFileMenuAction : AnAction() {
     override fun actionPerformed(e: AnActionEvent) {
-        val psiFile = e.getData(LangDataKeys.PSI_FILE) ?: return
-        val content = psiFile.text ?: return
-        val isVaulted = content.startsWith("\$ANSIBLE_VAULT")
-        val title = "${if (isVaulted) "Decrypt" else "Encrypt"} file with ansible-vault"
-
-        if (isVaulted) {
-            runTask(psiFile, title, DecryptFileAnsibleVaultRunnable(psiFile.project, psiFile, content))
-        } else {
-            val vaultIdentities = AnsibleConfigurationService.getInstance(psiFile.project)
-                .getAggregatedConfig()
-                .vaultIdentities
-
-            if (vaultIdentities != null && vaultIdentities.isNotEmpty()) {
-                AnsibleVaultIdentityPopup(vaultIdentities) {
+        val project = e.project ?: return
+        e.getData(LangDataKeys.VIRTUAL_FILE_ARRAY)?.forEach {
+            val psiFile = PsiManager.getInstance(project).findFile(it) ?: return@forEach
+            val content = psiFile.text ?: return@forEach
+            val isVaulted = content.startsWith("\$ANSIBLE_VAULT")
+            val title = "${if (isVaulted) "Decrypt" else "Encrypt"} file(s) with ansible-vault"
+
+            if (isVaulted) {
+                runTask(psiFile, title, DecryptFileAnsibleVaultRunnable(psiFile.project, psiFile, content))
+            } else {
+                val vaultIdentities = AnsibleConfigurationService.getInstance(psiFile.project)
```

Comment:
suggestion: Move into method like `#vaultFiles()`

The if block contains a bit too much nested logic for me. Splitting it into a separate method reduces cognitive load